plugins/modules/purefa_realm.py

```diff
@@ -0,0 +1,484 @@
+        )
+        if res.status_code != 200:
+            module.fail_json(
+                msg="Creation of realm {0} failed. Error: {1}".format(
+                    module.params["name"], res.errors[0].message
+                )
+            )
+    else:
+        res = array.post_realms(names=[module.params["name"]])
+        if res.status_code != 200:
+            module.fail_json(
+                msg="Creation of realm {0} failed. Error: {1}".format(
+                    module.params["name"], res.errors[0].message
+                )
+            )
+    if module.params["bw_qos"] and not module.params["iops_qos"]:
+        if int(human_to_bytes(module.params["bw_qos"])) in range(1048576, 549755813888):
+            changed = True
+            if not module.check_mode:
+                res = array.patch_realms(
+                    names=[module.params["name"]],
+                    realm=RealmPatch(bandwidth_limit=module.params["bw_qos"]),
+                )
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Setting of realm {0} QoS failed. Error {1}".format(
+                            module.params["name"],
+                            res.errors[0].name,
+                        )
+                    )
+        else:
+            module.fail_json(
+                msg="Bandwidth QoS value {0} out of range.".format(
+                    module.params["bw_qos"]
+                )
+            )
+    elif module.params["iops_qos"] and not module.params["bw_qos"]:
+        if int(human_to_real(module.params["iops_qos"])) in range(100, 100000000):
+            changed = True
+            if not module.check_mode:
+                res = array.patch_realms(
+                    names=[module.params["name"]],
+                    realm=RealmPatch(iops_limit=module.params["iops_qos"]),
+                )
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Setting of realm {0} QoS failed. Error {1}".format(
+                            module.params["name"],
+                            res.errors[0].name,
+                        )
+                    )
+        else:
+            module.fail_json(
+                msg="IOPs QoS value {0} out of range.".format(module.params["iops_qos"])
+            )
+    elif module.params["iops_qos"] and module.params["bw_qos"]:
+        bw_qos_size = int(human_to_bytes(module.params["bw_qos"]))
+        if int(human_to_real(module.params["iops_qos"])) in range(
+            100, 100000000
+        ) and bw_qos_size in range(1048576, 549755813888):
+            changed = True
+            if not module.check_mode:
+                res = array.patch_realms(
+                    names=[module.params["name"]],
+                    realm=RealmPatch(
+                        iops_limit=module.params["iops_qos"],
```

Comment:
iops_limit & bandwidth_limit should be in dict format & it should be assigned to 'qos' key.
This is leading to attribute error, should be corrected in all occurences of iops_limit & bandwidth_limit.
Eg: 
"qos": {
"bandwidth_limit": 1048576,
"iops_limit": 100
}